parastell/parastell.py

```diff
@@ -607,32 +591,80 @@ def parastell():
 
     vmec_file = all_data['vmec_file']
 
-    if args.logger == True:
-        logger = log.init()
-    else:
-        logger = log.NullLogger()
-
     stellarator = Stellarator(
         vmec_file,
         logger=logger
     )
 
-    build_full_model(
-        stellarator,
-        all_data['invessel_build'],
-        all_data['magnet_coils'],
-        all_data['source_mesh']
-    )
+    if args.ivb:
+        invessel_build = all_data['invessel_build']
+        stellarator.construct_invessel_build(**invessel_build)
+        stellarator.export_invessel_build(
+            export_dir=args.export_dir,
+            **(filter_kwargs(invessel_build, ivb.export_allowed_kwargs))
+        )
 
-    export_full_model(
-        stellarator,
-        args.export_dir,
-        all_data['invesel_build'],
-        all_data['magnet_coils'],
-        all_data['source_mesh'],
-        all_data['dagmc_export']
+    if args.magnets:
+        magnet_coils = all_data['magnet_coils']
+        stellarator.construct_magnets(**magnet_coils)
+        stellarator.export_magnets(
+            export_dir=args.export_dir,
+            **(filter_kwargs(magnet_coils, mc.export_allowed_kwargs))
     )
+
+    if args.source:
+        source_mesh = all_data['source_mesh']
+        stellarator.construct_source_mesh(**source_mesh)
+        stellarator.export_source_mesh(
+            export_dir=args.export_dir,
+            **(filter_kwargs(source_mesh, sm.export_allowed_kwargs))
+        )
     
+    if args.ivb or args.magnets:
+        dagmc_export = all_data['dagmc_export']
+        stellarator.export_dagmc(
+            export_dir=args.export_dir,
+            **dagmc_export
+        )
+
+    if args.nwl:
+        if not args.ivb:
+            invessel_build = all_data['invessel_build']
+            if not args.magnets:
+                dagmc_export = all_data['dagmc_export']
+        
+        if cubit_io.initialized:
+            cubit.cmd('new')
+
+        nwl_geom = Stellarator(
+            vmec_file,
+            logger=logger
+        )
+
+        nwl_construction_forbidden_kwargs = [
+            'plasma_mat_tag', 'sol_mat_tag', 'chamber_mat_tag',
+            'export_cad_to_dagmc', 'dagmc_filename'
+        ]
+        nwl_build = invessel_build
+        nwl_build['radial_build'] = {}
+        nwl_build['split_chamber'] = False
+        for name in nwl_build.keys() & nwl_construction_forbidden_kwargs:
+            del nwl_build[name]
+
+        nwl_geom.construct_invessel_build(**nwl_build)
+        nwl_geom.export_invessel_build(export_dir=args.export_dir)
+        
+        nwl_export_allowed_kwargs = [
+            'legacy_faceting', 'faceting_tolerance', 'length_tolerance',
+            'normal_tolerance', 'anisotropic_ratio', 'deviation_angle'
+        ]
+        nwl_geom.export_dagmc(
+            skip_imprint=True,
+            filename='nwl_geom',
+            export_dir=args.export_dir,
+            **(filter_kwargs(dagmc_export, nwl_export_allowed_kwargs))
```

Comment:
Removing this for simplicity
```suggestion
            export_dir=args.export_dir
```